neural_compressor/adaptor/torch_utils/gptq.py

```diff
@@ -0,0 +1,643 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import time
+import torch
+import torch.nn as nn
+import transformers
+from tqdm import tqdm
+from functools import partial
+from ...utils import logger
+
+DEBUG = False 
+
+# ==============model structure related==============
+def is_leaf(module):
+    """Judge whether a module has no child-modules.
+
+    Args:
+        module: torch.nn.Module
+
+    Returns:
+        a bool: whether a module has no child-modules.
+    """
+    children_cnt = 0
+    for n in module.children():
+        children_cnt += 1
+    return True if children_cnt == 0 else False
+
+def trace_gptq_target_blocks(module, module_types = [torch.nn.ModuleList]):
+    """Search transformer stacked structures, which is critical in LLMs and GPTQ execution.
+
+    Args:
+        module: torch.nn.Module
+        module_types: List of torch.nn.Module.
+
+    Returns:
+        gptq_related_blocks = {
+            "embeddings": {}, # Dict embedding layers before transfromer stack module, 
+            "transformers_pre": {}, # TODO
+            "transformers_name": string. LLMs' transformer stack module name ,
+            "transformers": torch.nn.ModuleList. LLMs' transformer stack module,
+            "transformers": {}, Dict# TODO
+        }
+    """
+    gptq_related_blocks = {
+        "embeddings": {},
+        "transformers_pre": {}, # todo
+        #"transformers_name": None, # None
+        #"transformers": None, # None
+        "transformers_name": "", # None
+        "transformers": [], # None
+        "transformers_post": {}, # todo
+    }
+    for n, m in module.named_modules():
+        if type(m) in module_types:
+            gptq_related_blocks["transformers_name"] = n
+            gptq_related_blocks["transformers"] = m
+            return gptq_related_blocks
+        else:
+            if is_leaf(m):
+                gptq_related_blocks["embeddings"][n] = m
+    return gptq_related_blocks
+
+def find_layers(module, layers=[nn.Conv2d, nn.Conv1d, nn.Linear, transformers.Conv1D], name=''):
+    """Get all layers with target types."""
+    if type(module) in layers:
+        return {name: module}
+    res = {}
+    for name1, child in module.named_children():
+        res.update(find_layers(
+            child, layers=layers, name=name + '.' + name1 if name != '' else name1
+        ))
+    return res
+
+def find_layers_name(module, layers=[nn.Conv2d, nn.Conv1d, nn.Linear, transformers.Conv1D], name=''):
+    """Get all layers with target types."""
+    if type(module) in layers:
+        return [name]
+    res = []
+    for name1, child in module.named_children():
+        res += find_layers_name(child, layers=layers, name = name + '.' + name1 if name != '' else name1)
+    return res
+
+def log_quantizable_layers_per_transformer(
+        transformer_blocks, 
+        layers=[nn.Conv2d, nn.Conv1d, nn.Linear, transformers.Conv1D]

```

Comment:
Since we normally could not get performance gain for conv2d, better to remove this one from the default setting unless users explicitly set it